frontend/src/metabase/collections/components/ActionMenu/ActionMenu.tsx

```diff
@@ -1,20 +1,47 @@
-import React, { useCallback } from "react";
+import React, { useCallback, useEffect, useState } from "react";
 
 import { ANALYTICS_CONTEXT } from "metabase/collections/constants";
 import { Item, Collection, isItemPinned } from "metabase/collections/utils";
 import EventSandbox from "metabase/components/EventSandbox";
 
 import { EntityItemMenu } from "./ActionMenu.styled";
+import { Bookmarks } from "metabase-types/api/bookmark";
 
 type Props = {
+  bookmarks?: Bookmarks;
+  createBookmark?: (id: string, collection: string) => void;
+  deleteBookmark?: (id: string, collection: string) => void;
   className?: string;
   item: Item;
   collection: Collection;
   onCopy: (items: Item[]) => void;
   onMove: (items: Item[]) => void;
 };
 
-function ActionMenu({ className, item, collection, onCopy, onMove }: Props) {
+function getIsBookmarked(item: Item, bookmarks: Bookmarks) {
+  return bookmarks.some(
+    bookmark => bookmark.type === item.model && bookmark.item_id === item.id,
+  );
+}
+
+function ActionMenu({
+  bookmarks,
+  createBookmark,
+  deleteBookmark,
+  className,
+  item,
+  collection,
+  onCopy,
+  onMove,
+}: Props) {
+  const [isBookmarked, setIsBookmarked] = useState(
```

Comment:
I don't understand why you would need `useState` + `useEffect` instead of `const isBookmarked = getIsBookmarked(item, bookmarks)`. Could you clarify?

Reply:
@ranquild Yeah, maybe we don't. Nothing is updating state after all, let me change this.